### PR TITLE
fix missing toolbar separator

### PR DESCRIPTION
**Pull request by yenda**
Tested on Android, waiting for build on iOS

Adds a 1px height separator between toolbar on view

Android screenshots
screenshot_1551973882
screenshot_1551973886
screenshot_1551973889
screenshot_1551973892
screenshot_1551973897
screenshot_1551973900
screenshot_1551973909
screenshot_1551973911
screenshot_1551973918
screenshot_1551973921
screenshot_1551973924
screenshot_1551973930
screenshot_1551973935
screenshot_1551973939
screenshot_1551973955
screenshot_1551973962
screenshot_1551973966
screenshot_1551973969
screenshot_1551974033
screenshot_1551975651
screenshot_1551973859
screenshot_1551973863
screenshot_1551973874
screenshot_1551973878

status: ready

### CODE REVIEW

**Review by rasom — APPROVED** [review]

**Comment by errorists** [comment]
@yenda is there an easy way for us change the separator colour on all Wallet screens (the ones using the blue background style) to 10% white, rgba(255,255,255,0.1)?

**Review by flexsurfer — APPROVED** [review]

**Comment by yenda** [comment]
@errorists yes I'll update the PR later

**Comment by vitvly** [comment]
iOS screenshots:

Browser: https://monosnap.com/file/LMouQGNWTW3sG3c2Ho7svbdMh41LjL
Browser edit: https://monosnap.com/file/csLAAli9Y6GO19ulWbPtHzBN2AQbp5
Public chat: https://monosnap.com/file/Fk1RiEbmnlvMwNhZPhEAyZKlSOjvny
Private chat: https://monosnap.com/file/iqPxTlogtt12GuL7YHy5dFi2q3181Y
Group chat 1: https://monosnap.com/file/eExG6LrYre6nvFQjfq3SSPvx38BY2Z
Group chat 2: https://monosnap.com/file/GnUDbemH2dgt5BhI0i1qSX0Wys8S6V
Group chat 3: https://monosnap.com/file/2s9yI6CtWI6DGOdOHdFZ7QqJKrBKIi
Profile: https://monosnap.com/file/anZjFeTjsK9Gk89N7LtINKx0iUhXku
QR code: https://monosnap.com/file/4Ol0TvmlCpf8dRh23bqUuF1Q75tq1X
Sign-in 1: https://monosnap.com/file/mqrojuW82NoUFRC4N6ijohGY6dRFUA
Sign-in 2: https://monosnap.com/file/cLjtdh3OqoKJKLAtcXIdqGRNu7bRmO
Recover account: https://monosnap.com/file/DcWq3DZc7D6jVL85cUtg7ntVywY29m

**Comment by errorists** [comment]
@siphiuel perfect, thanks! I take it all of the blue wallet views don't have separators like we just discussed right? On QR code and Recover account it looks like we have either a) two separators one above another b) a 2px high separator

**Comment by vitvly** [comment]
Updated:
QR code: https://monosnap.com/file/JdHZ5aNegvKO6qFuM5ABIVU7fpsWaK
Recover account: https://monosnap.com/file/9kCWe9hFQN23eg8yyNFtQ7rEzuyV3H
Wallet main: https://monosnap.com/file/z8vrYs0YFlliieQNTLuJzUJPyVmG5b
Wallet send: https://monosnap.com/file/qJiLrSjf36ID2JlJ1tc87NjaYcjSWU
Wallet receive: https://monosnap.com/file/DMHXaWMLgzcn9RSyRPZCDZzsGFKtDk
Wallet assets: https://monosnap.com/file/7KwPpGTgXZ3msEIcxOdaBY7mBjG14M

**Comment by errorists** [comment]
@siphiuel thanks, the separator is ✅ 
screenshot 2019-03-07 at 13 21 36
there's something weird happening here with alignment, the avatar looks like it's not centred. It should be 56 from left (if absolute) or 16 from <- icon and 10 from the top.

File: src/status_im/ui/screens/browser/styles.cljs
Comment (errorists) [inline src/status_im/ui/screens/browser/styles.cljs]: guys please, don't use letter-spacing property on text, thank you 🙏

Comment (yenda) [reply src/status_im/ui/screens/browser/styles.cljs]: never? there is 99 use of letter-spacing in the code base should I remove them all?

Comment (errorists) [reply src/status_im/ui/screens/browser/styles.cljs]: that would make my day

Comment (errorists) [reply src/status_im/ui/screens/browser/styles.cljs]: letter spacing was, kinda, required when using the system font on iOS to make it look native, iOS uses letter spacing in native apps. But we don't use system fonts anymore and we're not trying to imitate the exact native 'look' :)

Comment (errorists) [reply src/status_im/ui/screens/browser/styles.cljs]: or you can leave it for now, and return to it once we introduce text styles as variables

**Comment by errorists** [comment]
Noticed one more thing, in [private chats](
https://monosnap.com/file/iqPxTlogtt12GuL7YHy5dFi2q3181Y), the text properties for the username should be font-size: 15, font-weight: 500, line-height: 22, exactly same as for group or public chats, just without the subtitle.

**Comment by yenda** [comment]
@errorists should all be good now, check the screenshots for android I updated in the PR descriptions

**Comment by errorists** [comment]
sweet! looks good, you guys should give yourself a pat on the back

**Comment by churik** [comment]
@yenda 

1) "Next" button too close to keyboard (Sign in) :
sdfdsfdf
**Device:** Android 8, LG V20

2) "Back" is too close to input field and text in input field is not centered in editing mode 
wrwer
**Device:** Android 8, LG V20 (according text in input field)
"Back" is too close everywhere, doesn't depend on device.

3) Can't scroll to the bottom of contact list
default
**Device:** Android 8, LG V20

4) On iPhoneX (iOS 12.1.4) - there is a black space instead of currency section (doesn't matter if there is 0 USD or >0 USD value):
screenshot 2019-03-08 at 12 57 16
Reproducible also on IOS 11.4.1 (IPhone 7)

5) `c.call is not a function. (In 'c.call(null,s'), 'c.call' is undefined)` error when navigate back from 'Set up your wallet (the 3-words screen) to 'Let's get set up' screen. So New account -> Wallet tab -> tap `Let's get set up` -> tap top-left `<` button

screenshot 2019-03-08 at 12 57 26

6)  `<` back button is now instead of the `Done` button on Assets selection screen. @errorists what is better here to have: `Done` (like we had) or `<`?
screenshot 2019-03-08 at 13 16 06

7) Could we return back accessibility-id on closing QR code screen button screen, please?

screenshot 2019-03-08 at 13 42 22

8) Please rebase to recent develop, seems some issues with bottom panel already fixed in develop, but not in this PR

File: src/status_im/react_native/resources.cljs
Comment (errorists) [inline src/status_im/react_native/resources.cljs:65]: out of curiosity, why is the lock a png and not a vector file?

File: src/status_im/ui/components/connectivity/view.cljs
Comment (errorists) [inline src/status_im/ui/components/connectivity/view.cljs]: I think it should be a -3 like before, if that's the progress bar then it should placed be on top of the separator and cover it.

File: src/status_im/ui/components/toolbar/styles.cljs
Comment (errorists) [inline src/status_im/ui/components/toolbar/styles.cljs]: shouldn't it be 16? I've noticed it's 15 everywhere, are we subtracting the -1 for something?

File: src/status_im/ui/screens/browser/styles.cljs
Comment (errorists) [inline src/status_im/ui/screens/browser/styles.cljs]: are you guys sure it's 8? 56 total toolbar height - 36 url bar height / 2 gives 10

**Comment by errorists** [comment]
group
pls, can we fix the alignment here?

File: src/status_im/react_native/resources.cljs
Comment (yenda) [reply src/status_im/react_native/resources.cljs:65]: it's a complex image, I suppose it is the same as the tribute to talk one, in svg it wouldn't render properly?

File: src/status_im/ui/components/connectivity/view.cljs
Comment (yenda) [reply src/status_im/ui/components/connectivity/view.cljs]: so the separator isn't visible when the progress bar is there?

Comment (yenda) [reply src/status_im/ui/components/connectivity/view.cljs]: Screenshot_1552299886

Comment (yenda) [reply src/status_im/ui/components/connectivity/view.cljs]: Screenshot_1552300350

Comment (yenda) [reply src/status_im/ui/components/connectivity/view.cljs]: @errorists

**Comment by asemiankevich** [comment]
@yenda
9. "Backup your recovery phrase" on white background (possible for new users with  positive balance)

**OS:** IOS

10. Next button closes keyboard instead of redirecting to new step (can't reproduce on nightly 12/03/2019)

Steps: 
- open app
- tap on create new account
- enter password
- tap "Next"

**Expected:** will be redirected to next password field 
**Actual:** tap on next closes keyboard, need to tap once more to proceed
**OS:** IOS, Android

11. User input is partially cut in URL field in Status browser field 
input
**OS:** IOS

**Comment by yenda** [comment]
@asemiankevich thanks 9 and 10 should be fixed. 11 is normal, we only made the iOS button a little smaller to align it properly but it will be soon removed anyway. @errorists

**Comment by churik** [comment]
@yenda please, refresh link to get a comment updated) one more issue is found
Thanks for quick reaction!

**Comment by yenda** [comment]
@churik I made a change that might fix the new 11, will try on device once it is built

**Comment by yenda** [comment]
Ok I checked on iphone 6 plus and the url issue is gone @churik

**Comment by asemiankevich** [comment]
@errorists @yenda [Done] button is gone on Wallet -> Manage assets. However the assets selected / unselected are shown correctly to me even without this Done button. 

Do we need to have this button back here in this PR?

https://ethstatus.testrail.net/index.php?/tests/view/528248

**Comment by yenda** [comment]
@asemiankevich the new design is correct, so the test should be updated accordingly

**Comment by asemiankevich** [comment]
@antdanchenko please take this into account , we need to update tests ^

**Comment by antdanchenko** [comment]
I can't re-run the test since #7717 is not in the branch of the PR, please rebase the PR on latest develop if you want test to pass

however it's okay to merge the PR when testing is done, without re-running the test

**Comment by yenda** [comment]
@antdanchenko rebased

**Comment by yenda** [comment]
@asemiankevich @churik what do we do now, can we merge?

**Comment by annadanchenko** [comment]
on iOS build 25 https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7668/25/ on real iphone 5s:
issue 12. Backup recovery screen, button "Ok continue" is not visible. Part of text `recovery phrase` is hidden behind the image
pr

Note that on nightly build on same iphone 5s its half is visible, so user has possibility to tap on it.
IMG-4272

**Comment by annadanchenko** [comment]
> @asemiankevich @churik what do we do now, can we merge?

I've joined testing with iphone5s (smaller screen) and still testing, so might be more issues

**Comment by annadanchenko** [comment]
issue 13 It looks like tap area is smaller now on the buttons that sit on top area of the screen (like < ... +).  Usually it takes me 1-2 taps on < to return to Chats screen from chats or open mailservers list and then tap < to return. Now, I need to tap 3-4 times in order to get same result. @yenda can you check if tap area was decreased, please? @errorists can you try on your ios device to compare the experience with mine, please? If you are not focused on the tapping exactly on mid of the button and tap a bit on its right then nothing happens on my real iphone 5s...

Steps:
- open Profile -> Advanced -> Mailserver -> tap < to return to Profile
- open public chat -> tap < to return to Chats
- Open Wallet -> tap ... to open Manage Assets

**Comment by errorists** [comment]
touchable areas for all icons inside the toolbar should be a square 56px by 56px. When creating the Figma docs, we made the wrong assumption that this is a widely known practice, I've corrected it now for toolbars. 

Screenshot 2019-03-12 at 15 21 47

**Comment by asemiankevich** [comment]
@yenda issue 9 (white background for recovery phrase) is still reproducible for me in build 31. as well as for @annadanchenko

**Comment by vitvly** [comment]
Also added scroll view to Share profile screen and fixed margins there.

File: src/status_im/ui/components/connectivity/view.cljs
Comment (errorists) [reply src/status_im/ui/components/connectivity/view.cljs]: @yenda yeah but why is there a non transparent white background behind the progress bar? That's why I said to cover the separator, that it would still be visible.

**Comment by annadanchenko** [comment]
@yenda @siphiuel 
issue 12. Backup recovery screen, button "Ok continue" is not visible.  - fixed on ios in build 36 (can scroll and button is fully visible and can tap on it)

issue 9 - not fully fixed. white background is gone but previous element alignment (central) and background color (dark blue) are absent. Also there is no ">" button on the right side to show that it's tappable. compare now and nightly:

now (build 36 Android and iOS):
Screenshot_20190312-173334


nightly (ios)

**Comment by yenda** [comment]
@annadanchenko thanks I'll check again for 9

**Comment by yenda** [comment]
@annadanchenko @churik btw the seedphrase bubble in wallet will stay up no matter what, it's just to make testing easier, it's a separate commit I'll remove before merging

**Comment by annadanchenko** [comment]
@yenda @siphiuel  issue 14 No "..." button next to url field on Dapp screen. Build 41 Android and Build 42 iOS (most likely issues was there in older builds too)
Now:
Screenshot_20190313-102726

on nightly (expected)

**Comment by yenda** [comment]
@annadanchenko it's not in the new design
@errorists can you confirm?

**Comment by errorists** [comment]
Yes correct. The most up to date browser design we'll be following and implementing is [available here](https://www.figma.com/file/MYWmd1buvc2AMvUmFP9w42t5/Discovery?node-id=1182%3A5921)

However you guys need to sync up a bit with the browser team, the idea was that the share button would be moved from ••• and placed in the bottom toolbar, if we only remove it, then there's no way to access it from UI.

**Comment by yenda** [comment]
@errorists there is only a share link option behind the three dots, can I move it to the bottom bar from now? between chat and reload?

**Comment by errorists** [comment]
@yenda I pinged @flexsurfer about it and he agreed to do it as part of his PRs to separate Chats and Dapps, so it's between you guys who gets to work on it :)

**Comment by yenda** [comment]
@flexsurfer I'll just put the share icon to the bottom bar so that the option is still available in that PR

**Comment by annadanchenko** [comment]
PR looks ok to me now. things left: 
- check results of https://ci.status.im/job/end-to-end-tests/job/status-app-nightly/770/ (triggered nightly tests using this pr's apk to see if there are anything wrong from tests pov/features added/removed). 
- need feedback from Sergey with iphone X
- if above 2 are fine then we ask Eric to remove https://github.com/status-im/status-react/pull/7668#issuecomment-472327380 and check one more build once it's removed...

**Comment by Serhy** [comment]
I don't see any issues with iPhoneX (iOS 12.1.4) in that latest build https://i.diawi.com/MJzegm

**Comment by antdanchenko** [comment]
@yenda, please mind my commits before force pushing anything to the branch

**Comment by yenda** [comment]
@antdanchenko are you going to add some commits? I already squashed your commit related to e2e test

**Comment by antdanchenko** [comment]
@yenda ignore the comment above, I can't see my commit, however there are changes in e2e tests https://github.com/status-im/status-react/pull/7668/files#diff-154858b445ae2691db7259cc0e836b8c

**Comment by Serhy** [comment]
@yenda yes, yes one small change for tests we are going to merge. Should be quick.

**Comment by yenda** [comment]
@Serhy @antdanchenko you need to agree on something here :D

**Comment by Serhy** [comment]
@yenda autotests should be fixed.
We had failed autotests due to persistent 'Backup your recovery phrase' on Wallet screen which manually are fine (that is what should be removed according to https://github.com/status-im/status-react/pull/7668#issuecomment-472327380)

Could you please update the persistent 'Backup your recovery phrase' bubble change, so we could test the build before the merge?

**Comment by yenda** [comment]
@Serhy can you restart the e2e tests with latest build?

**Comment by Serhy** [comment]
> @Serhy can you restart the e2e tests with latest build?

Started nightly tests against latest build

**Comment by churik** [comment]
Tested on Android 8, LG V20 - retested all issues, mentioned in comments.
Last [build](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190313-140426-84426d-pr.apk) lgtm.
 Also briefly checked that change doesn't affect desktop Mac OS Build.
Good to go after rechecking nightly test results.

**Comment by yenda** [comment]
@annadanchenko @Serhy 100% e2e tests passed, is it good to merge?

**Comment by yenda** [comment]
@churik thanks!

**Pull request merged**
